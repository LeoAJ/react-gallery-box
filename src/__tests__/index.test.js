// @flow
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ReactGalleryBox from '../';

Enzyme.configure({ adapter: new Adapter() });

const images = [
  'url1',
  'url2',
  'url3'
];

test('should render component properly', () => {
  const component = shallow(<ReactGalleryBox images={images} />);
  expect(component).toMatchSnapshot();
});

test('should set state when mouse event triggered', () => {
  const component = shallow(<ReactGalleryBox images={images} />);
  component.simulate('mouseEnter');
  expect(component).toMatchSnapshot();
});
