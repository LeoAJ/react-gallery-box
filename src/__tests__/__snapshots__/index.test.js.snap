// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component properly 1`] = `
<div
  className="container"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="item"
    key="0"
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <img
      alt="url1"
      src="url1"
    />
  </div>
  <div
    className="item"
    key="1"
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <img
      alt="url2"
      src="url2"
    />
  </div>
  <div
    className="item"
    key="2"
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <img
      alt="url3"
      src="url3"
    />
  </div>
</div>
`;

exports[`should set state when mouse event triggered 1`] = `
<div
  className="container"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="item"
    key="0"
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <img
      alt="url1"
      src="url1"
    />
  </div>
  <div
    className="item"
    key="1"
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <img
      alt="url2"
      src="url2"
    />
  </div>
  <div
    className="item"
    key="2"
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <img
      alt="url3"
      src="url3"
    />
  </div>
</div>
`;
